logRequests.go

```diff
@@ -1,30 +1,71 @@
 package transport
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
 	"log"
 	"net/http"
 	"time"
 
 	"moul.io/http2curl/v2"
 )
 
-func LogRequests(next http.RoundTripper) http.RoundTripper {
-	return RoundTripFunc(func(req *http.Request) (resp *http.Response, err error) {
-		r := CloneRequest(req)
+type RequestLogger interface {
+	LogRequest(req *http.Request, curl *http2curl.CurlCommand)
```

Comment:
I'm not a fan of http2curl being part of any public interface

Reply:
Makes sense. Should I pass string directly ?